R/author.R

```diff
@@ -0,0 +1,139 @@
+#' `tar.gz` is built from package source.)
+
+#' * Presence of the fake author placed by [create_package()] and
+#' [use_description()]. This happens when \pkg{usethis} has to create a
+#' DESCRIPTION file and the user hasn't given any author information via the
+#' `fields` argument or the global option `"usethis.description"`. The
+#' placeholder looks something like `First Last <first.last@example.com> [aut,
+#' cre] (YOUR-ORCID-ID)` and `use_author()` offers to remove it in interactive
+#' sessions.
+#'
+#' @inheritParams utils::person
+#' @export
+#' @examples
+#' \dontrun{
+#' use_author(
+#'   given = "Lucy",
+#'   family = "van Pelt",
+#'   email = "lucy@example.com",
+#'   comment = c(ORCID = "LUCY-ORCID-ID")
+#' )
+#' }
+#'
+use_author <- function(given = NULL, family = NULL, ..., role = "ctb") {
+  check_is_package("use_author()")
+  maybe_name(given)
+  maybe_name(family)
+  check_character(role)
+
+  d <- proj_desc()
+  challenge_legacy_author_fields(d)
+  # We only need to consider Authors@R
+
+  author <- utils::person(given = given, family = family, role = role, ...)
+
+  authors_at_r_already <- d$has_fields("Authors@R")
+  if (authors_at_r_already) {
+    check_author_is_novel(given, family, d)
+  }
+  # This person is not already in Authors@R
+
+  if (authors_at_r_already) {
+    ui_done("
+      Adding to {ui_field('Authors@R')} in DESCRIPTION:
+      {format(author, style = 'text')}")
+  } else {
+    ui_done("
+      Creating {ui_field('Authors@R')} field in DESCRIPTION and adding:
+      {format(author, style = 'text')}")
+  }
+  d$add_author(given = given, family = family, role = role, ...)
+
+  challenge_default_author(d)
+
+  d$write()
+
+  invisible(TRUE)
+
+}
+
+challenge_legacy_author_fields <- function(desc = proj_desc()) {
+  has_legacy_field <-
+    desc$has_fields("Author") || desc$has_fields("Maintainer")
+  if (!has_legacy_field) {
+    return(invisible())
+  }
+
+  ui_oops("
+    Found legacy {ui_field('Author')} and/or {ui_field('Maintainer')} field \\
+    in DESCRIPTION.
+    usethis only supports modification of the {ui_field('Authors@R')} field.")
+  ui_info("
+    We recommend one of these paths forward:
+    * Delete these fields and rebuild with {ui_code('use_author()')}.
+    * Convert to {ui_field('Authors@R')} with {ui_code('desc::desc_coerce_authors_at_r()')},
+      then delete the legacy fields.")
+  if (ui_yeah("Do you want to cancel this operation and sort that out first?")) {
+    ui_stop("Cancelling.")
+  }
+  invisible()
+}
+
+check_author_is_novel <- function(given = NULL, family = NULL, d = proj_desc()) {
+  authors <- d$get_authors()
+  authors_given <- purrr::map(authors, "given")
+  authors_family <- purrr::map(authors, "family")
+  m <- purrr::map2_lgl(authors_given, authors_family, function(x, y) {
+    identical(x, given) && identical(y, family)
+  })
+  if (any(m)) {
+    aut_name <- glue("{given %||% ''} {family %||% ''}")
+    usethis_abort(c(
```

Comment:
usethis's UI needs an overhaul (pivoting towards cli), which I'm not necessarily biting off right now. But when we touch errors, we're phasing in `usethis_abort()`.

Reply:
For the overhaul, will the `ui_oops()`, `ui_info()` etc. be part of that? I.e., `usethis_abort()` is just the start?

Reply:
Yeah it will be everything. Some of those function will certainly live on, but their guts will be modified to use cli.